controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/EndpointCertificateMaintainer.java

```diff
@@ -93,17 +94,18 @@ private void updateRefreshedCertificates() {
     }
 
     /**
-     * If it's been four days since the cert has been refreshed, re-trigger all prod deployment jobs.
+     * If it's been four days since the cert has been refreshed, re-trigger prod deployment jobs (one at a time).
      */
     private void deployRefreshedCertificates() {
         var now = clock.instant();
+        var jobsTriggered = new AtomicInteger(0);
         curator.readAllEndpointCertificateMetadata().forEach((applicationId, endpointCertificateMetadata) ->
                 endpointCertificateMetadata.lastRefreshed().ifPresent(lastRefreshTime -> {
                     Instant refreshTime = Instant.ofEpochSecond(lastRefreshTime);
                     if (now.isAfter(refreshTime.plus(4, ChronoUnit.DAYS))) {
                         controller().applications().getInstance(applicationId)
                                 .ifPresent(instance -> instance.productionDeployments().forEach((zone, deployment) -> {
-                                    if (deployment.at().isBefore(refreshTime)) {
+                                    if (deployment.at().isBefore(refreshTime) && jobsTriggered.compareAndSet(0, 1)) {
```

Comment:
Perhaps randomise the order, lest a single, failing deployment blocks all others?

Reply:
Or better still, pick the eligible deployment with the oldest `lastTriggered` timestamp?

Reply:
That makes sense